recipes/vsg/all/conanfile.py

```diff
@@ -0,0 +1,126 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.microsoft import check_min_vs, is_msvc_static_runtime, is_msvc
+from conan.tools.files import get, copy, rm, rmdir, collect_libs
+from conan.tools.build import check_min_cppstd
+from conan.tools.scm import Version, Git
+from conan.tools.cmake import CMakeToolchain, CMakeDeps, CMake
+import os
+
+required_conan_version = ">=1.53.0"
+
+class VsgConan(ConanFile):
+    name = "vsg"
+    description = "VulkanSceneGraph"
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.vulkanscenegraph.org"
+    topics = ("vulkan", "scenegraph", "graphics", "3d")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "shader_compiler": [True, False],
+        "max_devices": [1,2,3,4],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "shader_compiler": True,
+        "max_devices" : 1,
+        "fPIC": True,
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 17
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "clang": "7",
+            "apple-clang": "10",
+        }
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def requirements(self):
+        self.requires("vulkan-loader/1.3.239.0", transitive_headers=True)
+
+    def validate(self):
+        if self.info.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        check_min_vs(self, 191)
+
+        if is_msvc_static_runtime(self):
+            raise ConanInvalidConfiguration(f"{self.name} does not support MSVC static runtime (MT/MTd) configurations, only dynamic runtime (MD/MDd) is supported")
+
+        if not is_msvc(self):
+            minimum_version = self._compilers_minimum_version.get(str(self.info.settings.compiler), False)
+            if minimum_version and Version(self.info.settings.compiler.version) < minimum_version:
+                raise ConanInvalidConfiguration(
+                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+                )
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+        if self.version in ("1.0.3"):
+            git = Git(self)
+            glslang_url = "https://github.com/vsg-dev/glslang.git"
+            glslang_branch = "VSG-1.0.x"
+            clone_args = ['--depth', '1', '--branch', glslang_branch]
+            git.clone(url= glslang_url, args=clone_args, target ="src/glslang")
```

Comment:
I cannot change how VSG is structured internally (it uses Cmake to pull the branch /formerly a submodule into its source tree). As you can see this is a fork of glslang and it is cloned into the sources of the VSG project. So I don't really understand how to change this to a dependency/requirement without potentially patching a lot of the VSG code.

If patching and using glslang as requirement is the only road here I'm afraid I won't be able to put the necessary effort into  this anytime soon. Apart from this, all other vsg-libraries use exactly the same approach. 
I'll take a dive into this option later this week, but I have the feeling that this might really complicate things in terms for  maintaining.

Do have any hints here? 

P.S. Thanks for your efforts

Reply:
Hi @psyinf thanks for the detailed response, it helps a lot! Let me talk this with the team to find a nice way forward :)

Reply:
How do we want to proceed with this? I can try to patch the vsg-cmake in a way it receives the original glslang dependencies for now. This will work for now, but won't help once vsg decides to depend on a custom fork. And as mentioned it might cause trouble later on in vsgImGui etc.

Reply:
@RubenRBS I've tried to patch it by using glslang as a dependency and failed. (The conan package for glslang doesn't seem to export the public headers used by VSG). So for now I'm not really sure if this is solvable at all without either 
a) dropping the support for shader compiling (I guess not really an option for the broader audience)
b) changing VSG itself (I had a hard time to get the author to recognize conan as a multiplicator for the library)

Reply:
The best approach would be creating a separated package for glslang, then, add it as build requirement (is that a build requirement only, right?). In Conan Center there is a package for glslang already available: https://conan.io/center/glslang, but it provided by Khronos group, I don't know how compatible is it, but you could create another package in case they are incompatible, like `vsg-glslang`.

Reply:
@uilianries  Thx for the insight. It is an optional build requirement (but without it, the library is practically severely hampered). The glsl-package on conan-center unfortunately is not an option as the vsg uses some parts that are not exported here. I'll consider the approach you presented by creating a vsg-glslang.
Can we for now simply cut this part out to finalize this PR? If this would be acceptable, I could remove the dependency for now, albeit only a temporary solution (I could make another attempt on integrating the glslang using your proposal later, when I find the time).

Reply:
@psyinf Yes, you can remove  glslang as dependency right now, and in a next PR, when `vsg-glslang` be available, you can add it as build requirement, no problem at all.

Reply:
@uilianries @RubenRBS  I've removed the glslang dependency handling for now and will follow your advice, hopefully in the near future. Please re-review and approve if you see fit

Reply:
@psyinf Thank you! Let's wait for the CI going back to normal, then we can review it again, but it's looking good.